modules/spring-config/src/main/java/org/mule/config/spring/SpringRegistryLifecycleManager.java

```diff
@@ -75,20 +111,33 @@ public void applyLifecycle(Object o) throws LifecycleException
      * {@link org.mule.config.spring.SpringRegistry#doDispose()} method which in turn
      * will destroy the application context managed by this registry
      */
-    class SpringContextDisposePhase extends ContainerManagedLifecyclePhase
+    class SpringContextDisposePhase extends MuleContextDisposePhase
     {
+
         public SpringContextDisposePhase()
         {
-            super(Disposable.PHASE_NAME, Disposable.class, Initialisable.PHASE_NAME);
-            registerSupportedPhase(NotInLifecyclePhase.PHASE_NAME);
-            // You can dispose from all phases
-            registerSupportedPhase(LifecyclePhase.ALL_PHASES);
+            super();
+            setIgnoredObjectTypes(new Class[] {
+                    Component.class,
+                    MessageSource.class,
+                    OutboundRouterCollection.class,
+                    OutboundRouter.class,
+                    Transformer.class,
+                    MuleContext.class,
+                    ServerNotificationManager.class});
         }
 
         @Override
         public void applyLifecycle(Object o) throws LifecycleException
         {
-            ((SpringRegistry) o).doDispose();
+            if (o instanceof SpringRegistry)
```

Comment:
This is horrible, but it is pre-existing and needs to be addressed with further iterations of registry lifecycle imporvements. Out of scope of this PR.